server/ServerApp.cpp

```diff
@@ -1638,6 +1638,42 @@ void ServerApp::AddObserverPlayerIntoGame(const PlayerConnectionPtr& player_conn
     // TODO: notify other players
 }
 
+void ServerApp::DropPlayerEmpireLink(int player_id)
+{ m_player_empire_ids.erase(player_id); }
+
+bool ServerApp::AddPlayerIntoGame(const PlayerConnectionPtr& player_connection) {
+    // search empire by player name
+    for (auto empire : Empires()) {
+        if (empire.second->PlayerName() == player_connection->PlayerName()) {
+            auto orders_it = m_turn_sequence.find(empire.first);
+            if (orders_it == m_turn_sequence.end()) {
+                WarnLogger() << "ServerApp::AddPlayerIntoGame empire " << empire.first
+                             << " for \"" << player_connection->PlayerName()
+                             << "\" doesn't wait for orders";
+                return false;
+            }
+            // make a link to new connection
+            m_player_empire_ids[player_connection->PlayerID()] = empire.first;
+
+            const OrderSet dummy;
+            const OrderSet& orders = orders_it->second ? *orders_it->second : dummy;
+
+            std::map<int, PlayerInfo> player_info_map = GetPlayerInfoMap();
+            bool use_binary_serialization = player_connection->ClientVersionStringMatchesThisServer();
+
+            player_connection->SendMessage(GameStartMessage(m_single_player_game, empire.first,
```

Comment:
bump the inner parameters to a new line... these are all a bit too indented to be easily readable

Reply:
Fixed.